src/server/pfs/fuse/loopback.go

```diff
@@ -538,16 +543,23 @@ func newLoopbackRoot(root, target string, c *client.APIClient, opts *Options) (*
 	}
 
 	n := &loopbackRoot{
-		rootPath:   root,
-		rootDev:    uint64(st.Dev),
-		targetPath: target,
-		write:      opts.getWrite(),
-		c:          c,
-		repoOpts:   opts.getRepoOpts(),
-		branches:   opts.getBranches(),
-		commits:    make(map[string]string),
-		files:      make(map[string]fileState),
-		stateMap:   make(map[string]string),
+		rootPath:    root,
+		rootDev:     uint64(st.Dev),
+		targetPath:  target,
+		write:       opts.getWrite(),
+		c:           c,
+		repoOpts:    opts.getRepoOpts(),
+		branches:    opts.getBranches(),
+		commits:     make(map[string]string),
+		files:       make(map[string]fileState),
+		stateMap:    make(map[string]string),
+		repoOpsCmts: make(map[string]string),
+	}
+
+	for _, opt := range opts.RepoOptions {
```

Comment:
Does the `repoOpsCmts` map store commit info not already in `repoOpts`?

Reply:
To my understanding, repoOpts is mapping a mount name to a repoOption. What we use repoOpsCmts here is to map the repo name to a commit ID.
Before this change: When resolving a commit id by mount name, the code first check if this mapping is already stored in the cache. If not, the code resolve the branch first and then use the branch name to inspectCommit and pick the head of that branch info. If branch cannot be resolved, "master" branch is used.
Since we nilled out branch names, we cannot use this way to resolve branch then pick the head of that branchInfo. We directly map the mount name to commitID, so that when the code resolves a commit ID it check this map first.

Reply:
> Can we use `repoOpts` instead to get the commit ID?

Let me think about it.

Reply:
There seem to be 2 codepaths that determine what information a `File` in a `RepoOption` has. One is if a user calls `pachctl mount` with [no repo/branch/commit arguments](https://github.com/pachyderm/pachyderm/blob/master/src/server/pfs/fuse/fuse.go#L28) and the other one is if [they do call it with arguments](https://github.com/pachyderm/pachyderm/blob/master/src/server/pfs/cmds/mount_unix.go#L77). In the case where no arguments are passed, the `File` is [created with a branch name and no commit ID](https://github.com/pachyderm/pachyderm/blob/master/src/server/pfs/fuse/fuse.go#L56). In the case where arguments are passed, it depends on [what format the user provides](https://github.com/pachyderm/pachyderm/blob/master/src/internal/cmdutil/cobra.go#L125). 

Based on this context, the `File` should either have a branch value, a commit ID value, or both. Thus, the check you have below (if both branch and commit ID are empty) makes sense.